californium-core/src/test/java/org/eclipse/californium/core/coap/RequestTest.java

```diff
@@ -67,8 +68,21 @@ public void testSetURISetsDestination() throws UnknownHostException {
 
 	@Test
 	public void testSetURISetsUriHostOption() throws UnknownHostException {
-		Request req = Request.newGet().setURI("coap://iot.eclipse.org");
-		assertThat(req.getOptions().getUriHost(), is("iot.eclipse.org"));
```

Comment:
As far I understand uri-host is used as 'host' for HTTP virtual host.
Using "localhost" as a virtual host sounds legit to me. I would remove the "localhost" case in setURI and change the test to use "localhost".
If you strictly apply the RFC I think it's incorrect to remove uri-host when it's value is "localhost"